src/App.jsx

```diff
@@ -1,128 +1,185 @@
+  script.onload = async () => {
+    console.log('=============> lib-jitsi-meet Loaded <=============');
+    JitsiMeetJS.init();
+    const configScript = document.createElement('script');
+    configScript.src = `https://${domain}/config.js`;
+    document.querySelector('head').appendChild(configScript);
+    configScript.onload = async () => {
+      console.log('=============> jitsi config Loaded <=============');
+      config.serviceUrl = config.websocket || config.bosh;
+      config.serviceUrl += `?room=${room}`;
+      const connection = new JitsiMeetJS.JitsiConnection(
+        null,
+        undefined,
+        config,
+      );
+      connection.addEventListener(
+        JitsiMeetJS.events.connection.CONNECTION_ESTABLISHED,
+        async () => {
           console.log('=============> CONNECTION ESTABLISHED <=============');
-          const conference = connection.initJitsiConference(room, {})
-          await conference.join()
+          const conference = connection.initJitsiConference(room, {});
+          await conference.join();
           console.log('=============> CONFERENCE JOINED <=============');
           const localTracks = await JitsiMeetJS.createLocalTracks(
             { devices: ['video', 'audio'], facingMode: 'user' },
             true,
           );
-          console.log('=============> Video & Audio Tracks Created <=============');
-          const localVideoTrack = localTracks.find((track) => track.getType() === 'video');
+          console.log(
+            '=============> Video & Audio Tracks Created <=============',
+          );
+          const localVideoTrack = localTracks.find(
+            (track) => track.getType() === 'video',
+          );
           const localAudioTrack = localTracks.find(
             (track) => track.getType() === 'audio',
           );
           conference.addTrack(localVideoTrack);
           conference.addTrack(localAudioTrack);
-          console.log('=============> Video & Audio Tracks Connected <=============');
-          resolve({conference, localVideoTrack})
-        })
-        await connection.connect()
-      }
+          console.log(
+            '=============> Video & Audio Tracks Connected <=============',
+          );
+          resolve({ conference, localVideoTrack });
+        },
+      );
+      await connection.connect();
     };
-  })
+  };
+});
+
+// JITSI STANDUP CUSTOM HOOK
+const useTracks = () => {
+  const [tracks, setTracks] = useState([]);
+
+  const addTrack = useCallback(
+    (track) => {
+      setTracks((tracks) => {
+        const hasTrack = tracks.find(
+          (_track) => track.getId() === _track.getId(),
+        );
+
+        if (hasTrack) return tracks;
+
+        return [...tracks, track];
+      });
+    },
+    [setTracks],
+  );
+
+  const removeTrack = useCallback(
+    (track) => {
+      setTracks((tracks) => tracks.filter((_track) => track.getId() !== _track.getId()));
+    },
+    [setTracks],
```

Comment:
I wonder why we put the setTracks function in the dependency array. When does this change? Or is that the idea? This will never change.